benchmarks/Makefile

```diff
@@ -0,0 +1,14 @@
+CXX=g++
+HDF5_ROOT=/usr/local
```

Comment:
Can't this use CMake, too?

Reply:
So move poster `.cxx` files into `examples`, and add the compilation bits?

Reply:
I'd replace the old ones, though and then of course there needs to be some update in the doc generation. If you think this is a good idea, I can take care of it.

Reply:
Thanks for the feedback. I see the benefits of recycling the benchmark demos to examples. My only two arguments against are:
 - Makefiles make it dead-easy to see and tune compilation flags. I was playing with optimization and inlining  levels, etc, without thinking "maybe cmake is trying to do smth smart".
 - Having a separate folder with the run_benchmarks script would help anybody to reproduce the results we got and eventually proceed from there.

Having that in mind I kind of would prefer just moving the benchmarks to src, but no strong feelings.

Reply:
Sorry to jump on this only now but have you considered using the Google benchmark library? It is very straightforward to use. On one side, it requires a new library to link against but on the other side, it would wrap the 2 executables and the bash script in one single executable, and would not require caliper.

Reply:
That sounds very interesting, will definitely read about it. Nevertheless, I would like to close this topic, maybe that's a nice improvement for a free-friday?